packages/analytics-adapter/src/data-layer-adapter.test.ts

```diff
@@ -0,0 +1,175 @@
+import {DataLayerAdapter} from "./data-layer-adapter";
+
+const synchronousContext = {
+    example: {
+        complex: 'item',
+        with: {
+            many: 'layers',
+            andTypes: 3
+        }
+    },
+    example2: true
+};
+
+const asynchronousContext1 = {
+    more: 'complex',
+    data: {
+        'like': 'this'
+    },
+    example: {
+        makeSureWeDeepMerge: true
+    }
+};
+
+const asynchronousContext2 = {
+    example: {
+        makeSureWeDeepMerge2: true
+    }
+};
+
+const interval1Multiplier = 3;
+const interval2Multiplier = 6;
```

Comment:
May want to point out that `interval2Multiplier` must be greater than `interval1Multiplier` based on usages below.